nano/test_common/testutil.hpp

```diff
@@ -35,12 +37,20 @@
 	GTEST_FATAL_FAILURE_)
 
 /** Asserts that the condition becomes true within the deadline */
-#define ASSERT_TIMELY(time, condition)    \
-	system.deadline_set (time);           \
-	while (!(condition))                  \
-	{                                     \
-		ASSERT_NO_ERROR (system.poll ()); \
-	}
+#define ASSERT_TIMELY(time, condition)      \
```

Comment:
The problem with this is that it checks the condition twice. It should only check it once per loop.

For example, it won't catch this condition:
```
bool counter_increment_and_equals_ten()
{
    static int i = 0;
    return (--i) == 10;
}